kratos/linear_solvers/amgcl_solver.h

```diff
@@ -51,423 +47,315 @@
 
+    /// The sparse matric type
     typedef typename TSparseSpaceType::MatrixType SparseMatrixType;
 
+    /// Vector type definition
     typedef typename TSparseSpaceType::VectorType VectorType;
 
+    /// Dense matrix type
     typedef typename TDenseSpaceType::MatrixType DenseMatrixType;
 
-    AMGCLSolver(Parameters rParameters)
-    {
+    /// DofArray type
+    typedef ModelPart::DofsArrayType DofsArrayType;
+
+    /// The index type definition
+    typedef std::size_t IndexType;
 
+    /// The size type definition
+    typedef std::size_t SizeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief This is the default constructor
+     * @param ThisParameters The configuration parameters
+     */
+    AMGCLSolver(Parameters ThisParameters = Parameters(R"({})"))
+    {
         Parameters default_parameters( R"(
-                                   {
-                                       "solver_type" : "AMGCL",
-                                       "smoother_type":"ilu0",
-                                       "krylov_type": "gmres",
-                                       "coarsening_type": "aggregation",
-                                       "max_iteration": 100,
-                                       "provide_coordinates": false,
-                                       "gmres_krylov_space_dimension": 100,
-                                       "verbosity" : 1,
-                                       "tolerance": 1e-6,
-                                       "scaling": false,
-                                       "block_size": 1,
-                                       "use_block_matrices_if_possible" : true,
-                                       "coarse_enough" : 5000
-                                   }  )" );
-
-
-        //now validate agains defaults -- this also ensures no type mismatch
-        rParameters.ValidateAndAssignDefaults(default_parameters);
-
-        //validate if values are admissible
-        std::set<std::string> available_smoothers = {"spai0","ilu0","ilut","iluk","damped_jacobi","gauss_seidel","chebyshev"};
-        std::set<std::string> available_solvers = {"gmres","bicgstab","cg","bicgstabl","lgmres", "bicgstab_with_gmres_fallback","idrs"};
+        {
+            "solver_type"                    : "AMGCL",
+            "smoother_type"                  :"ilu0",
+            "krylov_type"                    : "gmres",
+            "coarsening_type"                : "aggregation",
+            "max_iteration"                  : 100,
+            "provide_coordinates"            : false,
+            "gmres_krylov_space_dimension"   : 100,
+            "verbosity"                      : 1,
+            "tolerance"                      : 1e-6,
+            "scaling"                        : false,
+            "block_size"                     : 1,
+            "use_block_matrices_if_possible" : true,
+            "coarse_enough"                  : 5000
```

Comment:
actually we should reduce this to let's say 1000

Reply:
This was already like this, not my choice